pkg/auth/authorizer.go

```diff
@@ -156,15 +158,14 @@ func (m Authorizer) Authenticate(w http.ResponseWriter, r *http.Request, upstrea
 				w.Header().Add("Set-Cookie", cookie.Name+"=delete; path=/; expires=Thu, 01 Jan 1970 00:00:00 GMT")
 			}
 		}
-		if (!m.AuthRedirectDisabled)  {
+		if !m.AuthRedirectDisabled {
 			redirOpts := make(map[string]interface{})
 			redirOpts["auth_url_path"] = m.AuthURLPath
 			redirOpts["auth_redirect_query_disabled"] = m.AuthRedirectQueryDisabled
 			redirOpts["redirect_param"] = m.AuthRedirectQueryParameter
+			redirOpts["use_js_redir"] = m.UseJSRedir
 			//redirOpts["logger"] = m.logger
-			jwthandlers.AddRedirectLocationHeader(w, r, redirOpts)
-			w.WriteHeader(302)
-			w.Write([]byte(`Unauthorized`))
+			jwthandlers.HandleRedir(w, r, redirOpts)
```

Comment:
@caiych , let's do a condition check here, without wraping it all in one HandleRedir function, .e.g

```
if m. RedirectWithJavascript {
  jwthandlers.HandleJavascriptRedirect(w, r, redirOpts)
}
```

Reply:
Add a separate commit for this, frankly I don't think this is making things clearer.
1. the diversion repeats 4 times in package auth. (Is there a good reason for this?)
2. most of the redirecting logic is shared(most of the redirecting logic is about finding the right URL)
3. there are early returns in the shared logic, making the shared function messy.

What I think might be slightly better(than the original commit in this PR) would be at the end of the HandleRedir function, calling a different function(so the intention is clearer in function name) based on the redirOpts.

Reply:
Wait, the test passed after this commit??